src/Microsoft.AspNetCore.Razor.Evolution/IDirectiveDescriptorBuilder.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright(c) .NET Foundation.All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    public interface IDirectiveDescriptorBuilder
+    {
+        IDirectiveDescriptorBuilder AddType();
+
+        IDirectiveDescriptorBuilder AddMember();
+
+        IDirectiveDescriptorBuilder AddString();
+
+        IDirectiveDescriptorBuilder AddLiteral(string literal, bool optional);
```

Comment:
Don't love the "optional" part of this but this comes into play when doing `@addTagHelper "..."` where the quotes are optional. Also works for ending optional semicolons which exist on many of our directives.

If you can think of something better I'm all ears 😄